sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -730,13 +730,16 @@ const assetLockExpirationCallback = (
             if (evaluateError) {
                 // If it succeeds, no change
                 const [reclaimResult, isReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
-                if (isReclaimableQueryError) {
-                    let i = 0;
-                    do {
-                        reclaimAssetInHTLC(contract, assetType, assetID, recipientECert);
-                        i++;
-                    } while (i < 3);                
-                }
+                let i = 0;
+                do {
+                    if (isReclaimableQueryError) {
```

Comment:
This is not what I meant.
If condition should be the outside the loop only.
But when you call the function in next line (736), that should use same logic as in line 732:
`let [retryReclaimResult, retryReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));`
`if (!retryReclaimableQueryError) { break }`

i.e. when you re-try, break out of loop when the reclaim is successful, it doesn't make sense to retry 3 times even though first retry succeeded.